extension/icu/icu-strptime.cpp

```diff
@@ -293,7 +293,7 @@ struct ICUStrptime : public ICUDateFunc {
 
 				    // Change TZ if one was provided.
 				    if (tz.GetSize()) {
-					    SetTimeZone(calendar, tz);
+					    SetTimeZone(calendar, tz, parameters.error_message);
```

Comment:
In case of an error we should call `HandleCastError` similar to the code above, instead of (still) returning a timestamp. Could you add a test that triggers this with an explicit `TRY_CAST` and an invalid timezone?